subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformArtifactOperation.java

```diff
@@ -16,15 +16,19 @@
 
 package org.gradle.api.internal.artifacts.transform;
 
+import com.google.common.base.Preconditions;
+import org.gradle.api.NonNullApi;
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ResolvableArtifact;
 import org.gradle.internal.operations.BuildOperationContext;
-import org.gradle.internal.operations.RunnableBuildOperation;
 import org.gradle.internal.operations.BuildOperationDescriptor;
+import org.gradle.internal.operations.RunnableBuildOperation;
 
+import javax.annotation.Nullable;
 import java.io.File;
 import java.util.List;
 
-class TransformArtifactOperation implements RunnableBuildOperation {
+@NonNullApi
+class TransformArtifactOperation implements RunnableBuildOperation, ArtifactTransformationResult {
```

Comment:
This is a bit confusing for me, the operation and its result being implemented by the same type.